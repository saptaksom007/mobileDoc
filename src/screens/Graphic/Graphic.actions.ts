export const ActionTypes = {
  GRAPHIC_REQUESTED: 'GRAPHIC_REQUESTED',
  GRAPHIC_SUCCEEDED: 'GRAPHIC_SUCCEEDED',
  GRAPHIC_FAILED: 'GRAPHIC_FAILED',
}

export default class Actions {
  static graphicRequested(surveyId: number, patientUuid: string) {
    return {
      type: ActionTypes.GRAPHIC_REQUESTED,
      payload: { surveyId, patientUuid },
    }
  }
  static graphicSucceeded(data: any[]) {
    return {
      type: ActionTypes.GRAPHIC_SUCCEEDED,
      payload: data,
    }
  }
  static graphicFailed(error: Error) {
    return {
      type: ActionTypes.GRAPHIC_FAILED,
      payload: {
        error,
      },
    }
  }
}
