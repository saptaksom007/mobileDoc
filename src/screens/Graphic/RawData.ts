export interface RawData {
  date: number
  value: number
}
