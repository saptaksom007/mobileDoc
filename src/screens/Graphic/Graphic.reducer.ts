import { ActionTypes } from './Graphic.actions'

const { GRAPHIC_REQUESTED, GRAPHIC_SUCCEEDED, GRAPHIC_FAILED } = ActionTypes

interface State {
  data: any[]
}

interface Action {
  type: keyof typeof ActionTypes
  payload: any
}

const initialState: State = {
  data: [],
}

export default function graphic(state: State = initialState, action: Action) {
  switch (action.type) {
    case GRAPHIC_REQUESTED: {
      return { ...state }
    }

    case GRAPHIC_SUCCEEDED: {
      return { ...state, data: action.payload }
    }

    case GRAPHIC_FAILED: {
      return { ...state }
    }

    default:
      return state
  }
}
