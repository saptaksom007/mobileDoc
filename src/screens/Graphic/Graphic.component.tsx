import * as React from 'react'
import i18n from 'ex-react-native-i18n'
import {
  View,
  StyleSheet,
  Dimensions,
  FlatList,
  TextProps,
  TouchableWithoutFeedback,
} from 'react-native'
import { connect } from 'react-redux'
import Svg, {
  Path,
  Defs,
  LinearGradient,
  Stop,
  Circle,
  G,
} from 'react-native-svg'
import { scaleTime, scaleLinear } from 'd3-scale'
import * as shape from 'd3-shape'
import Axis from './components/Axis'
import { Color } from 'constants/Color'
import { Small } from 'components/CustomText/CustomText.component'
import { RawData } from './RawData'
import { useCallbackOne } from 'use-memo-one'
import {
  calculateLineHeight,
  SMALL_SIZE,
} from 'components/CustomText/CustomText.styles'
import { momentWithLocales } from 'utilities/date'

const { width: windowWidth, height: windowHeight } = Dimensions.get('window')

const phi = 1.618
const height = (1 - 1 / phi) * windowHeight
const strokeWidth = 2
const legendHeightBase = (windowHeight - height) / 4
const graphPadding = 17

const ITEM_PADDING_VERTICAL = 10

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    width: windowWidth,
    height: windowHeight,
  },
  item: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
    padding: ITEM_PADDING_VERTICAL,
    backgroundColor: 'white',
  },
  label: {
    color: Color.black,
  },
})

const Separator = () => (
  <View
    style={{
      width: '100%',
      backgroundColor: Color.tintColorTTT,
      height: 1,
    }}
  />
)
const Label = ({
  style,
  ...props
}: TextProps & { children: React.ReactNode }) => (
  <Small style={[styles.label, style]} {...props} />
)

const HeaderList = () => (
  <View style={styles.item}>
    <Label>{i18n.t('surveys.graphic.date')}</Label>
    <Label>{i18n.t('surveys.graphic.weight')}</Label>
  </View>
)

const Item = (date?: number, onPress?: (i: RawData) => void) => ({
  item,
}: {
  item: RawData
}) => {
  const selected = date === item.date
  const selectedColor = [selected ? { color: Color.white } : undefined]
  return (
    <TouchableWithoutFeedback onPress={() => onPress!(item)}>
      <View
        style={[
          styles.item,
          selected ? { backgroundColor: Color.blue } : undefined,
        ]}
      >
        <Label style={selectedColor}>
          {momentWithLocales(item.date).format('DD MMM')}
        </Label>
        <Label style={selectedColor}>{item.value} Kg</Label>
      </View>
    </TouchableWithoutFeedback>
  )
}

interface Props {
  data: RawData[]
}

const ITEM_HEIGHT = ITEM_PADDING_VERTICAL * 2 + calculateLineHeight(SMALL_SIZE)
export const Graphic = ({ data }: Props) => {
  const [point, setPoint] = React.useState<RawData | undefined>()
  const { scaleX, scaleY, xAxisScale, yAxisScale } = createScales(
    data,
    windowWidth,
    height,
    graphPadding,
  )
  const d = shape
    .line<RawData>()
    .x(p => scaleX(p.date))
    .y(p => scaleY(p.value))
    .curve(shape.curveLinear)(data) as string
  const keyExtractor = ({ date }: any) => `${date}`
  const flatListRef = React.useRef<FlatList<RawData>>()
  const selectDot = useCallbackOne(
    (item: RawData) => () => {
      if (flatListRef.current) {
        setPoint(item)
        flatListRef.current.scrollToIndex({
          index: data.findIndex((i: RawData) => i.date === item.date),
          animated: true,
        })
      }
    },
    [],
  )
  const itemLayout = useCallbackOne(
    (_data: any, index: number) => ({
      length: ITEM_HEIGHT,
      offset: ITEM_HEIGHT * index,
      index,
    }),
    [data],
  )
  return (
    <View style={styles.container}>
      <View
        style={{
          width: windowWidth,
          height,
          backgroundColor: 'white',
          overflow: 'hidden',
        }}
      >
        <Svg
          height={height}
          width={windowWidth}
          viewBox={`0 0 ${windowWidth} ${height}`}
        >
          {data && data.length > 1 ? (
            <>
              <Defs>
                <LinearGradient
                  id='gradient'
                  x1='50%'
                  y1='0%'
                  x2='50%'
                  y2='100%'
                >
                  <Stop offset='0%' stopColor={Color.gradientBlue} />
                  <Stop offset='80%' stopColor={Color.gradientBlueL} />
                  <Stop offset='100%' stopColor={Color.gradientBlueLL} />
                </LinearGradient>
              </Defs>
              <Path
                d={`${d}L ${windowWidth} ${height} L 0 ${height}`}
                fill='url(#gradient)'
              />
            </>
          ) : null}

          <Axis
            width={windowWidth - 2 * graphPadding}
            x={graphPadding}
            y={height - graphPadding}
            ticks={8}
            scale={xAxisScale}
          />
          <Axis
            width={height - 2 * graphPadding}
            vertical
            x={graphPadding}
            y={height - graphPadding}
            ticks={8}
            scale={yAxisScale}
          />
          <Path
            fill='transparent'
            stroke={Color.tintColor}
            {...{ d, strokeWidth }}
          />
          {data.map(({ value, date }, i) => (
            <G key={`${date}-${i}`}>
              <Circle
                stroke={point?.date === date ? Color.tintColor : Color.white}
                strokeWidth={3}
                pointerEvents='none'
                cx={scaleX(date)}
                cy={scaleY(value)}
                r={5}
                fill={point?.date === date ? Color.white : Color.tintColor}
              />
              <Circle
                onPress={selectDot({ value, date })}
                cx={scaleX(date)}
                cy={scaleY(value)}
                r={12}
                fill={'rgba(0,0,0,0)'}
              />
            </G>
          ))}
        </Svg>
      </View>
      <View
        style={{
          width: windowWidth,
          paddingHorizontal: 40,
          paddingTop: 15,
          height: legendHeightBase * 3,
          backgroundColor: Color.white,
        }}
      >
        <FlatList
          ref={flatListRef as any}
          showsVerticalScrollIndicator={false}
          ItemSeparatorComponent={Separator}
          ListHeaderComponent={HeaderList}
          stickyHeaderIndices={[0]}
          getItemLayout={itemLayout}
          {...{ keyExtractor, data, renderItem: Item(point?.date, setPoint) }}
        />
      </View>
    </View>
  )
}
const DAY_IN_MS = 24 * 60 * 60 * 1000
const getDomain = (domain: number[]) => [
  Math.min(...domain),
  Math.max(...domain),
]

const createScales = (
  dataPoints: RawData[],
  width: number,
  gHeight: number,
  padding: number,
) => {
  const onlyOne = dataPoints.length === 1
  const onlyOneDate = dataPoints[0].date
  const dateTimes = getDomain(
    !onlyOne
      ? dataPoints.map(pair => pair.date)
      : [
          onlyOneDate - DAY_IN_MS * 4,
          onlyOneDate - DAY_IN_MS * 3,
          onlyOneDate - DAY_IN_MS * 2,
          onlyOneDate - DAY_IN_MS,
          onlyOneDate,
          onlyOneDate + DAY_IN_MS,
          onlyOneDate + DAY_IN_MS * 2,
          onlyOneDate + DAY_IN_MS * 3,
          onlyOneDate + DAY_IN_MS * 4,
        ],
  )
  const values = getDomain(
    !onlyOne
      ? dataPoints.map(pair => pair.value)
      : [
          dataPoints[0].value / 3,
          dataPoints[0].value + dataPoints[0].value / 3,
        ],
  )

  const scaleX = scaleTime()
    .domain(dateTimes)
    .range([padding, width - padding])

  const scaleY = scaleLinear()
    .domain(values)
    .range([gHeight - padding, padding])

  const xAxisScale = scaleTime()
    .domain([padding, width - padding])
    .range(dateTimes)

  const yAxisScale = scaleLinear()
    .domain([gHeight - padding, padding])
    .range(values)

  return { scaleX, scaleY, xAxisScale, yAxisScale }
}

function select({ graphic: { data } }: any): object {
  return { data }
}

export default connect(select)(Graphic)
