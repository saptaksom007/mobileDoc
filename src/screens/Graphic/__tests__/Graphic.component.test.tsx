import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { Graphic } from '../Graphic.component'

it('<Graphic/> render correctly', () => {
  const tree = renderer
    .create(<Graphic data={[{ date: 1583249764616, value: 1 }]} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
