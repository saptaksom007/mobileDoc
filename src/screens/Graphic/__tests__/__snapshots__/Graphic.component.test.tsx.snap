// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Graphic/> render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "height": 1334,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": 509.5253399258345,
        "overflow": "hidden",
        "width": 750,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={509.5253399258345}
      bbWidth={750}
      focusable={false}
      height={509.5253399258345}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 509,
            "width": 750,
          },
        ]
      }
      vbHeight={509.5253399258345}
      vbWidth={750}
      width={750}
    >
      <RNSVGGroup>
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={17}
            x2={733}
            y1={492.5253399258345}
            y2={492.5253399258345}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={17}
            x2={17}
            y1={492.5253399258345}
            y2={502.7539113544059}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={119}
            x2={119}
            y1={492.5253399258345}
            y2={502.7539113544059}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={221}
            x2={221}
            y1={492.5253399258345}
            y2={502.7539113544059}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={323}
            x2={323}
            y1={492.5253399258345}
            y2={502.7539113544059}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={425}
            x2={425}
            y1={492.5253399258345}
            y2={502.7539113544059}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={527}
            x2={527}
            y1={492.5253399258345}
            y2={502.7539113544059}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={629}
            x2={629}
            y1={492.5253399258345}
            y2={502.7539113544059}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={731}
            x2={731}
            y1={492.5253399258345}
            y2={502.7539113544059}
          />
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                17,
              ]
            }
            y={
              Array [
                512.9824827829773,
              ]
            }
          >
            <RNSVGTSpan
              content="28 Feb"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                119,
              ]
            }
            y={
              Array [
                512.9824827829773,
              ]
            }
          >
            <RNSVGTSpan
              content="01 Mar"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                221,
              ]
            }
            y={
              Array [
                512.9824827829773,
              ]
            }
          >
            <RNSVGTSpan
              content="02 Mar"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                323,
              ]
            }
            y={
              Array [
                512.9824827829773,
              ]
            }
          >
            <RNSVGTSpan
              content="03 Mar"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                425,
              ]
            }
            y={
              Array [
                512.9824827829773,
              ]
            }
          >
            <RNSVGTSpan
              content="04 Mar"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                527,
              ]
            }
            y={
              Array [
                512.9824827829773,
              ]
            }
          >
            <RNSVGTSpan
              content="05 Mar"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                629,
              ]
            }
            y={
              Array [
                512.9824827829773,
              ]
            }
          >
            <RNSVGTSpan
              content="06 Mar"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                0,
                3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                731,
              ]
            }
            y={
              Array [
                512.9824827829773,
              ]
            }
          >
            <RNSVGTSpan
              content="07 Mar"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={17}
            x2={17}
            y1={492.5253399258345}
            y2={17}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={10.206780858202364}
            x2={400}
            y1={492.5253399258345}
            y2={492.5253399258345}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={10.206780858202364}
            x2={400}
            y1={425.5253399258345}
            y2={425.5253399258345}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={10.206780858202364}
            x2={400}
            y1={358.5253399258345}
            y2={358.5253399258345}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={10.206780858202364}
            x2={400}
            y1={291.5253399258345}
            y2={291.5253399258345}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={10.206780858202364}
            x2={400}
            y1={224.5253399258345}
            y2={224.5253399258345}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={10.206780858202364}
            x2={400}
            y1={157.5253399258345}
            y2={157.5253399258345}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={10.206780858202364}
            x2={400}
            y1={90.52533992583449}
            y2={90.52533992583449}
          />
          <RNSVGLine
            propList={
              Array [
                "stroke",
                "strokeWidth",
              ]
            }
            stroke={3642452638}
            strokeWidth="1"
            x1={10.206780858202364}
            x2={400}
            y1={23.525339925834487}
            y2={23.525339925834487}
          />
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                -3,
                -3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                3.413561716404729,
              ]
            }
            y={
              Array [
                492.5253399258345,
              ]
            }
          >
            <RNSVGTSpan
              content="0"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                -3,
                -3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                3.413561716404729,
              ]
            }
            y={
              Array [
                425.5253399258345,
              ]
            }
          >
            <RNSVGTSpan
              content="0"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                -3,
                -3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                3.413561716404729,
              ]
            }
            y={
              Array [
                358.5253399258345,
              ]
            }
          >
            <RNSVGTSpan
              content="1"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                -3,
                -3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                3.413561716404729,
              ]
            }
            y={
              Array [
                291.5253399258345,
              ]
            }
          >
            <RNSVGTSpan
              content="1"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                -3,
                -3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                3.413561716404729,
              ]
            }
            y={
              Array [
                224.5253399258345,
              ]
            }
          >
            <RNSVGTSpan
              content="1"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                -3,
                -3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                3.413561716404729,
              ]
            }
            y={
              Array [
                157.5253399258345,
              ]
            }
          >
            <RNSVGTSpan
              content="1"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                -3,
                -3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                3.413561716404729,
              ]
            }
            y={
              Array [
                90.52533992583449,
              ]
            }
          >
            <RNSVGTSpan
              content="1"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
          <RNSVGText
            content={null}
            dx={Array []}
            dy={Array []}
            fill={4280431936}
            font={
              Object {
                "fontFamily": "System",
                "fontSize": "9",
                "fontWeight": "500",
                "textAnchor": "middle",
              }
            }
            matrix={
              Array [
                1,
                0,
                0,
                1,
                -3,
                -3,
              ]
            }
            propList={
              Array [
                "fill",
              ]
            }
            rotate={Array []}
            x={
              Array [
                3.413561716404729,
              ]
            }
            y={
              Array [
                23.525339925834487,
              ]
            }
          >
            <RNSVGTSpan
              content="1"
              dx={Array []}
              dy={Array []}
              font={Object {}}
              rotate={Array []}
              x={Array []}
              y={Array []}
            />
          </RNSVGText>
        </RNSVGGroup>
        <RNSVGPath
          d="M375,175.5084466419448Z"
          fill={0}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={3642452638}
          strokeWidth={2}
        />
        <RNSVGGroup>
          <RNSVGCircle
            cx={375}
            cy={175.5084466419448}
            fill={3642452638}
            pointerEvents="none"
            propList={
              Array [
                "fill",
                "stroke",
                "strokeWidth",
              ]
            }
            r={5}
            stroke={4294967295}
            strokeWidth={3}
          />
          <RNSVGCircle
            cx={375}
            cy={175.5084466419448}
            fill={0}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            propList={
              Array [
                "fill",
              ]
            }
            r={12}
            responsible={true}
          />
        </RNSVGGroup>
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "height": 618.3559950556241,
        "paddingHorizontal": 40,
        "paddingTop": 15,
        "width": 750,
      }
    }
  >
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      ListHeaderComponent={[Function]}
      data={
        Array [
          Object {
            "date": 1583249764616,
            "value": 1,
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      getItemLayout={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={
        Array [
          0,
        ]
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flexDirection": "row",
                "justifyContent": "space-around",
                "padding": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 17.5,
                    "lineHeight": 22,
                    "overflow": "visible",
                  },
                  Object {
                    "color": "rgba(27, 106, 158, 0.85)",
                    "fontSize": 15,
                    "lineHeight": 18,
                  },
                  Array [
                    Object {
                      "color": "#363c40",
                    },
                    undefined,
                  ],
                ]
              }
            >
              surveys.graphic.date
            </Text>
            <Text
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 17.5,
                    "lineHeight": 22,
                    "overflow": "visible",
                  },
                  Object {
                    "color": "rgba(27, 106, 158, 0.85)",
                    "fontSize": 15,
                    "lineHeight": 18,
                  },
                  Array [
                    Object {
                      "color": "#363c40",
                    },
                    undefined,
                  ],
                ]
              }
            >
              surveys.graphic.weight
            </Text>
          </View>
        </View>
        <View
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                  "padding": 10,
                },
                undefined,
              ]
            }
          >
            <Text
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 17.5,
                    "lineHeight": 22,
                    "overflow": "visible",
                  },
                  Object {
                    "color": "rgba(27, 106, 158, 0.85)",
                    "fontSize": 15,
                    "lineHeight": 18,
                  },
                  Array [
                    Object {
                      "color": "#363c40",
                    },
                    Array [
                      undefined,
                    ],
                  ],
                ]
              }
            >
              03 Mar
            </Text>
            <Text
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "TitilliumWeb-Regular",
                    "fontSize": 17.5,
                    "lineHeight": 22,
                    "overflow": "visible",
                  },
                  Object {
                    "color": "rgba(27, 106, 158, 0.85)",
                    "fontSize": 15,
                    "lineHeight": 18,
                  },
                  Array [
                    Object {
                      "color": "#363c40",
                    },
                    Array [
                      undefined,
                    ],
                  ],
                ]
              }
            >
              1
               Kg
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
